Updated Readme with usage instructions (#15)
Babylonpartners/Stevenson: README.md

# iOS Build Distribution System

`Stevenson` is a Vapor framework designed to build integrations between Slack apps, Github, JIRA and CI services (CircleCI).
This project also contains implementation of the Slack app used by Babylon iOS team (if you want to know more about how our team works checkout our [playbook](https://github.com/Babylonpartners/ios-playbook))

## 🚀 Usage

To use `Stevenson` in your app add it as a dependency to your `Package.swift`:

```swift
dependencies: [
    .package(url: "https://github.com/Babylonpartners/Stevenson.git", .branch("master")),
]
```

and then import it to your project:

```swift
import Stevenson
```

### Supported services

`Stevenson` comes with implementation of Slack [slash commands](https://api.slack.com/slash-commands), GitHub, JIRA and CircleCI APIs. At the moment it does not implement complete set of these APIs but only provides bare minimum required for the functionality of the app. 

> To create these services use corrseponding types constructors providing required values. It's adviced but not required to store these values in the environment variables:

Reviewer: ```suggestion
To create these services use corresponding type constructors providing required values. It's advised but not required to store these values in the environment variables:
```

  --data 'token=__SLACK_TOKEN__&channel_name= __SLACK_CHANNEL__&text=somelane%20someargs'
```

## 🕹 Adding a new Slack command to the app

If you need to create a new Slack command:

 1. Go to the Slack Commands config page for your team's Slack app: `https://api.slack.com/apps/<YourSlackAppID>/slash-commands`
 2. Click on "Create New Command"
   * Fill in the slash command (e.g. `/foo`)
   * Enter `https://<appname>.herokuapp.com/<command>` as the request URL, replacing `<appname>` with the name of your Heroku app instance (e.g. `stevenson-bot`) and `<command>` by the command name (e.g. `foo`)
   * Fill in the short description and the hint for the command
   * Hit "Save"
 3. Open the project in Xcode and add a new handler for the Slack command:
   * Open `commands.swift`
   * Implement your command as a static function in the `SlackCommand` namespace:
   
   ```swift
extension SlackCommand {

>    static let <command> = { 

Reviewer: ```suggestion
   static func <command>(/* optional params if needed */) { 
```

>         SlackCommand(

Reviewer: ```suggestion
        return SlackCommand(
```

            name: "<command>", 
            help: "...",
            allowedChannels: [...],
            run: { ... }
        ) 

>     }

Reviewer: This example seems like a mix of a static property and a static func 😅 

Since the rest of the doc below assumes it was declared as a function (`.<command>()`), better be consistent here then. (I've added 2 separate "GitHub suggestions" above since we can't make a GitHub suggestion on multiple lines at once, but those would need to be applied together to make sense)

Author: We currently have these commands defined as static closures, so I'd prefer to keep this example close to how it is in the implementation
